src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const resultArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (stateCopy[key]) {
+            delete stateCopy[key];
+          }
+        }
```

Comment:
-Interesting solution here. You used 'for...of' for the object, and its works.
I'm confused about how it's working.

Reply:
I diced to use this for loop because `actions` is an array of objects and with arrays we use `for of`
hope I didn`t do it wrong)

Reply:
`action.keysToRemove`, guys? it's an array, handle with for-of-loop